frontend/src/components/sign-up/sign-up.tsx

```diff
@@ -1,99 +1,109 @@
+import { useForm, FieldValues } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
+import logo from 'assets/img/logo-dark.svg';
 import { UserCreatePayload } from 'common/types/types';
 import {
+  AppRoute,
   ButtonType,
   DataStatus,
+  InputPlaceholder,
   InputType,
+  LabelNames,
   UserCreatePayloadKey,
 } from 'common/enums/enums';
+import { SignupSchema } from 'validation-schemas/validation-schemas';
+import { useAppSelector, useDispatch } from 'hooks/hooks';
 import { auth as authActions } from 'store/actions';
-import { useAppSelector, useDispatch, useState } from 'hooks/hooks';
-import { Input, Button } from 'components/common/common';
+import { Button, Input, Link } from 'components/common/common';
 import { DEFAULT_REGISTER_PAYLOAD } from './common/constants';
+import styles from './styles.module.scss';
 
 const SignUp: React.FC = () => {
-  const [registerPayload, setRegisterPayload] = useState<UserCreatePayload>(
-    DEFAULT_REGISTER_PAYLOAD,
-  );
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FieldValues>({
+    defaultValues: DEFAULT_REGISTER_PAYLOAD,
+    resolver: joiResolver(SignupSchema),
+    mode: 'onSubmit',
+  });
+
   const { authStatus } = useAppSelector(({ auth }) => ({
     authStatus: auth.dataStatus,
   }));
+
   const dispatch = useDispatch();
 
   const isFormDisable = authStatus === DataStatus.PENDING;
 
-  const handleSubmit = (evt: React.FormEvent): void => {
-    evt.preventDefault();
-    dispatch(authActions.signUp(registerPayload));
-  };
-
-  const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>): void => {
-    setRegisterPayload({
-      ...registerPayload,
-      [target.name]: target.value,
-    });
+  const onSubmit = (data: UserCreatePayload): void => {
+    dispatch(authActions.signUp(data));
   };
 
   return (
-    <section>
-      <h1>Sign Up ♊️</h1>
-
-      <form onSubmit={handleSubmit}>
-        <p>
+    <div className={styles.signUpPage}>
+      <form className={styles.form} onSubmit={handleSubmit(onSubmit)}>
+        <img
+          src={logo}
+          className={styles.formLogo}
+          loading="lazy"
+          alt="Jabber logo"
+        />
+        <h3 className={styles.formTitle}>Sign Up</h3>
+        <div className={styles.formSubtitle}>
+          Already have an account?
+          <Link to={AppRoute.SIGN_IN}>Sign In</Link>
```

Comment:
```suggestion
          <Link to={AppRoute.SIGN_IN} className={styles.link}>Sign In</Link>
```
prettify it

Reply:
If this link is located in common components, we don't need styles in this file attached to it